docs/concepts/resources.md

```diff
@@ -0,0 +1,17 @@
+Resources
+=========
+
+A resource is extra data used by the game, and is stored in a data file, instead of being in the code. Resources are contained in the `assets` directory on the classpath. In the default mod development kit, this directory is under the `src/main/resources` directory of the project. They include things such as models, textures, and localization files.
+
+When multiple resource packs are enabled, they are merged. Generally, files from resource packs at the top of the stack override those below; however, for certain files, such as localization files, they are actually merged contentwise. Mods actually define resource packs too, and their resources are simply merged with the default resource pack inside the core Minecraft JAR in the same way as any other.
+
+All resources should have snake_case paths and domains; furthermore, lowercasing is enforced in 1.11 and above. This casing convention is used even outside the resource system, generally whenever a `String`/`ResourceLocation` identifier is needed (e.g. modids). (This does not mean Java identifiers should be snake cased.)
```

Comment:
The second sentence seems almost unnecessary. I'm not even sure what it's trying to say, really. Perhaps this bit of info could be merged somewhere else?